lib/Graph/Log.cpp

```diff
@@ -86,60 +87,67 @@ void LogContext::logNodeCreation(const Node *newNode) {
-    logStr.append(") <== ");
+    addLogContent(
+        llvm::formatv(
+            "[FULL SCOPE: {0}] --- CREATE { (Kind: {1}, Name: {2}) <== ",
+            getFullScopeName(), newNode->getKindName(), newNode->getName())
+            .str());
     for (auto n : inputs) {
-      logStr.append(" (Kind: ");
-      logStr.append(n->getKindName());
-      logStr.append(", Name: ");
-      logStr.append(n->getName());
-      logStr.append(") ");
+      addLogContent(llvm::formatv(" (Kind: {0}, Name: {1}) ", n->getKindName(),
+                                  n->getName())
+                        .str());
     }
-    logStr.append(" }\n");
-    addLogContent(logStr);
+    addLogContent(" }\n");
   }
 }
 
 /// Logs the node replacement.
 void LogContext::logNodeReplacement(const Node *oldNode, const Node *newNode) {
   if (dumpCompilationLogOpt) {
-    std::string logStr;
-    logStr.append("[FULL SCOPE: ");
-    logStr.append(getFullScopeName());
-    logStr.append(" ]");
-    logStr.append(" --- REPLACE { (Kind: ");
-    logStr.append(oldNode->getKindName());
-    logStr.append(", Name: ");
-    logStr.append(oldNode->getName());
-    logStr.append(") <== (Kind: ");
-    logStr.append(newNode->getKindName());
-    logStr.append(", Name: ");
-    logStr.append(newNode->getName());
-    logStr.append(") }\n");
-    addLogContent(logStr);
+    addLogContent(llvm::formatv("[FULL SCOPE: {0} ] --- REPLACE ( (Kind: {1}, "
+                                "Name: {2}) <==  (Kind: {3}, Name: {4}) }\n",
+                                getFullScopeName(), oldNode->getKindName(),
+                                oldNode->getName(), newNode->getKindName(),
+                                newNode->getName())
+                      .str());
   }
 }
 
 /// Logs the node deletion.
 void LogContext::logNodeDeletion(const Node &deletedNode) {
   if (dumpCompilationLogOpt) {
-    std::string logStr;
-    logStr.append("[FULL SCOPE: ");
-    logStr.append(getFullScopeName());
-    logStr.append(" ]");
-    logStr.append(" --- DELETE { (Kind: ");
-    logStr.append(deletedNode.getKindName());
-    logStr.append(", Name: ");
-    logStr.append(deletedNode.getName());
-    logStr.append(") }\n");
-    addLogContent(logStr);
+    addLogContent(llvm::formatv("[FULL SCOPE: {0} ] --- DELETE ( (Kind: {1}, "
+                                "Name: {2}) }\n",
+                                getFullScopeName(), deletedNode.getKindName(),
+                                deletedNode.getName())
+                      .str());
+  }
+}
+
+/// Logs node's input changes.
+void LogContext::logNodeInputChange(const Node *user, const Node *prevOpr,
+                                    const Node *newOpr) {
+  if (dumpCompilationLogOpt) {
+    addLogContent(
+        llvm::formatv(
+            "[FULL SCOPE: {0} ] --- NODE INPUT CHANGE { User(Kind: {1}, "
+            "Name: {2}) :: ",
+            getFullScopeName(), user->getKindName(), user->getKindName())
```

Comment:
I think you meant to get the name of the user for the last argument here, i.e. : `getFullScopeName(), user->getKindName(), user->getName())`